users/resources.py

```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class StudentPrizesResource(APIView):
+    serializer_class = PrizeSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request, pk):
+        user_id = request.user.id
+
+        if not has_user_access_to_student(user_id, pk):
```

Comment:
It's not always necessary to declare variable if we're using it only at one place. As an alternative we could write it as

```suggestion
        if not has_user_access_to_student(request.user.id, pk):
```

but I'll leave the decision up to you